### PR TITLE
Increase test coverage for ListUtils

### PR DESCRIPTION
**Pull request by sigee**
- isEqualList: same size, but different elements in the lists
 - isEqualList: same size, same elements, but different order
 - hashCodeForList: with null element
 - partition.get: edge cases (Exception)
 - union: multiple cases
 - sum: happy path

### CODE REVIEW

File: src/test/java/org/apache/commons/collections4/ListUtilsTest.java [hunk 1]
```diff
@@ -81,16 +81,25 @@ public void testEmptyIfNull() {
...
-        final List<String> a = new ArrayList<>( data );
-        final List<String> b = new ArrayList<>( data );
-
-        assertEquals(a, b);
-        assertTrue(ListUtils.isEqualList(a, b));
-        a.clear();
-        assertFalse(ListUtils.isEqualList(a, b));
-        assertFalse(ListUtils.isEqualList(a, null));
-        assertFalse(ListUtils.isEqualList(null, b));
+        final List<String> list1 = new ArrayList<>( data );
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/collections4/ListUtilsTest.java:84]: Please don't rename variables. I can't tell what's changed.

Comment (sigee) [reply src/test/java/org/apache/commons/collections4/ListUtilsTest.java:84]: I renamed them because they overlapped the String constants from lines 46, 47. And they were confusing.
```
private static final String a = "a";
private static final String b = "b";
```
```
final List<String> a = new ArrayList<>( data );
final List<String> b = new ArrayList<>( data );
```

**Comment by sigee** [comment]
I split the changes into two commits to make them more understandable.
- Increase the coverage
- Rename local variables

**Comment by garydgregory** [comment]
@sigee 
Merged. TY.
